### PR TITLE
Nexus interleave

### PR DESCRIPTION
**Pull request by biologyguy**
This pull request supersedes pull request #362

The original pull request seems stuck, because @theboocock has not replied to requests to accept the license agreement and incorporate suggestions by @PamelaM.

This new request is a modification of 362, in that it implements a full override mechanism instead of a partial (negative only) override. A full override allows users to force interleave alignments where 70<columns<1000

**************
I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

I am happy be thanked by name in the ``NEWS.rst`` and ``CONTRIB.rst`` files,
and have added myself to those files as part of this pull request. (*This
acknowledgement is optional. Note we list the names sorted alphabetically.*)

### CODE REVIEW

**Comment by peterjc** [comment]
@biologyguy there are merge conflicts - are you happy to rebase this onto the latest master branch? Something like this:

```
git checkout nexus_interleave  # if not already there
git fetch origin
git rebase origin/master
```

If the conflicts prove to tricky, ``git rebase --abort`` should get you back to how things are now. If it works, then you'd need to force push the changes to update this pull request (one of the few times you would normally git ``git push --force ...`` as it rewrites git history).

**Comment by peterjc** [comment]
Note to self: Steve is already in ``CONTRIB.rst`` and in the ``NEWS.rst`` file for the next release.

**Comment by biologyguy** [comment]
I still don't understand rebase well enough to make it work when it isn't working (it wasn't)... I went with a hard reset instead. Is okay now?

File: Bio/AlignIO/NexusIO.py [hunk 1]
```diff
@@ -115,11 +120,11 @@ def write_alignment(self, alignment):
...
             n.add_sequence(record.id, str(record.seq))
-        # For smaller alignments, don't bother to interleave.
-        # For larger alginments, interleave to avoid very long lines
-        # in the output - something MrBayes can't handle.
-        # TODO - Default to always interleaving?
-        n.write_nexus_data(self.handle, interleave=(columns > 1000))
+
+        # Note: MrBayes may choke on large alignments if not interleaved
+        if interleave is None:
+            interleave = False if columns <= 1000 else True
```
Comment (peterjc) [inline Bio/AlignIO/NexusIO.py]: How about simplifying that to just:

```
if interleave is None:
    interleave = (columns > 1000)
```

File: Tests/test_Nexus.py [hunk 2]
```diff
@@ -363,6 +363,47 @@ def test_NexusTest1(self):
...
+    def test_write_alignment():
+        # Default causes no interleave (columns <= 1000)
+        records = [SeqRecord(Seq("ATGCTGCTGA" * 90, alphabet=ambiguous_dna), id=_id) for _id in ["foo", "bar", "baz"]]
+        a = MultipleSeqAlignment(records, alphabet=ambiguous_dna)
+
+        handle = StringIO()
+        NexusWriter(handle).write_alignment(a)
+        handle.seek(0)
+        data = handle.read()
+        assert "ATGCTGCTGA" * 90 in data
```
Comment (peterjc) [inline Tests/test_Nexus.py]: Can you switch that ``assert`` statement (and the similar ones below) to:

```python
self.assertIn("ATGCTGCTGA" * 90, data)
```

This is new in Python 2.7, prior to that we'd have used a workaround like ``self.assertTrue("ATGCTGCTGA" * 90 in data)`` instead.

**Comment by peterjc** [comment]
However you did it with git, the changes and history looks clear now. I noted some minor style issues, but otherwise this looks good (assuming the automated tests are still happy).

**Comment by biologyguy** [comment]
Seems like TravisCI and Codecov are cool with the changes :)

**Comment by peterjc** [comment]
Merged, thank you.

**Pull request merged**